### ISSUE TITLE
Determine space savings of unused files directories for deleted multisites.

### ISSUE THREAD

**Issue by richardbporter** [uiowa/uiowa#4086]
### Problem
Not much disk space left. :)

Multisite files directories are symlinked to another location, e.g. `docroot/sites/mysite.com/files` actually points to `/mnt/gfs/app.env/sites/mysite.com`. This means none of those files directories get removed when deleting multisites.

### Proposed solution
For each application environment, SSH in and run `diff /mnt/gfs/app.env/sites/ sites/` where `app.env` is the application machine name and environment, ex. `uiowa702.test`. The directories that are listed as only being in the `/mnt/gfs/...` directory can be deleted because they has no matching multisite. That is, the files directory was never deleted during a previous deprovision for thsoe sites.

The above process could probably be automated.

**Comment by richardbporter** [comment]
This really only applies to large multisite applications like uiowa701/2/3. A potential first slice could be to get the diff of files directories that can go away and then use another bash script to get the amount of space that would be saved by deleting those.

Probably makes sense to do #3953 first.

**Comment by richardbporter** [comment]
### Proposed solution

- Determine how much total space each application filesystem is taking up with `ncdu`. Comment here. 
- Get the unused multisite directories for each application using the `diff` command.
- Pipe the unused multisite directories into `du -sh --total dir1 dir2` to get the total. Comment here.

**Comment by richardbporter** [comment]
https://support.acquia.com/hc/en-us/articles/360005257274-Assessing-disk-space-usage-on-Acquia-Cloud-Classic

**Comment by richardbporter** [comment]
@mark-bennett-uiowa and @pyrello: Are there any multisite directories on clas and grad that could be deleted now because they've relaunched in SNv3 (or gone away)? That would get them to show up as unused here (and reduce our misc. D7 site counts!). 

If you proceed, don't forget to delete the corresponding database. :)

**Comment by pyrello** [comment]
@richardbporter We are identifying the sites that can be de-provisioned currently and will proceed with that in the coming weeks.

**Comment by richardbporter** [comment]
TLDR;
I think there is about 60gb of files that could be deleted.

I'm working on a folder in O365 within CSI-Drupal that will detail more of this.

**Comment by richardbporter** [comment]
https://iowa.sharepoint.com/:f:/r/sites/CSI-Drupal2/Shared%20Documents/Acquia%20File%20Usage?csf=1&web=1&e=D1x6f7

**Comment by richardbporter** [comment]
The assumption that these unused multisite files directories can be removed is based on:
1. the symlink to the corresponding files directory has not been changed.
  - Ex. docroot/sites/mysite.com/files -> /mnt/gfs/app/sites/someothersite.com 
2. the public files path for the site has not been changed to another site.
  - Ex. mysite.com public files path is set to sites/someothersite.com/files

I don't think the first is possible. AC creates the symlinks automatically and I can't delete it on dev. There could be a situation where one multisite is symlinked to another as described in https://docs.acquia.com/cloud-platform/manage/files/symlinks/ but I doubt that is common. The second probably isn't common either but could happen maybe with a URL change. Unless AC sets that automatically - might be the case in D9/BLT but not D7.

I'm fairly confident neither of these things was done on uiowa701/2/3 but less confident on the other applications. In any case, I think we could transfer all these files directories to O365 before deleting them (compressed).

**Comment by joewhitsitt** [comment]
i don't know if this was back from on-prem days but i do recall ip.uiowa.edu/international.uiowa.edu kind of difference related to files.

**Comment by richardbporter** [comment]
Yea thanks good call. I'm not sure we can easily check for those types of things. I would say we probably have to plan on backing up the files before deleting. :(
